Tests/ttLib/ttFont_test.py

```diff
@@ -0,0 +1,44 @@
+import io
+from fontTools.ttLib import TTFont, newTable, registerCustomTableClass, unregisterCustomTableClass
+from fontTools.ttLib.tables.DefaultTable import DefaultTable
+
+
+class CustomTableClass(DefaultTable):
+
+    def decompile(self, data, ttFont):
+        self.numbers = list(data)
+
+    def compile(self, ttFont):
+        return bytes(self.numbers)
+
+    # not testing XML read/write
+
+
+table_C_U_S_T_ = CustomTableClass  # alias for testing
+
+
+TABLETAG = "CUST"
+
+
+def test_registerCustomTableClass():
+    font = TTFont()
+    font[TABLETAG] = newTable(TABLETAG)
+    font[TABLETAG].data = b"\x00\x01\xff"
+    f = io.BytesIO()
+    font.save(f)
+    f.seek(0)
+    assert font[TABLETAG].data == b"\x00\x01\xff"
+    registerCustomTableClass(TABLETAG, "ttFont_test", "CustomTableClass")
```

Comment:
maybe wrap the block between register/unregister in a try... finally clause.
Or if we find ourselves wanting to be able to temporarily register/unregister custom classes outside of self-testing, we could turn the registerCustomTableClass into a context manager that automatically deregisters itself upon exit the `with` block. E.g. you wrap it with contextlib.contextmanager decorator, do the global registering then do a bare `yield` and finally the unregistering. If one doesn't care about the final part, one can call the function as usual and ignore the return value

Reply:
I literally only added unregister because it messed up the tests in unpredictable ways :) I think normal use cases wouldn't want to unregister ever, so a context manager seems overkill to me.

That said, I'll add the try/except to the tests. I considered and then was too lazy. But you're right, it's better.

Reply:
(Btw. simply calling a `contextlib.contextmanager`-wrapped function doesn't execute the pre-yield bit, so for the suggester behavior a context manager class would be needed, with the registry being done in `__init__` instead of `__enter__`.)

Reply:
ah ok thanks for the checking. Probably not worth making it a full context manager class.